src/test/java/htsjdk/samtools/BAMFileWriterTest.java

```diff
@@ -229,4 +237,61 @@ public void testBAMReadBases() throws IOException {
         Assert.assertEquals(recordFromBAM.getReadBases(), SequenceUtil.toBamReadBasesInPlace(originalSAMRecord.getReadBases()));
     }
 
+
+    @DataProvider
+    public Object[][] longCigarsData(){
+        return new Object[][]{
+                {1},
+                {10},
+                {100},
+                {1_000},
+                {10_000},
+                {100_000},
+                {1_000_000}};
+    }
+
+    @Test(dataProvider = "longCigarsData")
+    public void testLongCigars(int numOps) throws Exception {
+        final SAMRecordSetBuilder builder = new SAMRecordSetBuilder(true, SAMFileHeader.SortOrder.coordinate);
+
+        final List<CigarOperator> operators = new ArrayList<>(numOps);
+        final CigarOperator operatorstoUse[] = new CigarOperator[]{CigarOperator.M, CigarOperator.D, CigarOperator.M, CigarOperator.I};
+        for (int i = 0; i < numOps; i++) {
+            operators.add(operatorstoUse[i % operatorstoUse.length]);
+        }
+        operators.add(CigarOperator.M);
+
+        final Cigar cigar = Cigar.fromCigarOperators(operators);
+
+        builder.addFrag("frag1",0,1,false,false,cigar.toString(),null,30);
+        builder.addPair("pair1",0,1,100_000,false,false,cigar.toString(),cigar.toString(),true,false,30);
+
+        testHelper(builder, SAMFileHeader.SortOrder.coordinate,true);
+    }
+
+
+    //why is this not breaking?
+    @Test(dataProvider = "longCigarsData")
```

Comment:
do you understand, @lh3 , why this test isn't failing?! 

I"m adding a fake CG tag and I have long cigars, but somehow it manages to encode and decode the read correctly....I was expecting failure!